### ISSUE TITLE
Configure/Enable Gii 

### ISSUE THREAD

**Issue by zeroth** [yiisoft/yii2-gii#128]
Hello There, 
I want to user Gii in my Yii app and I am getting error 
```
Invalid Parameter – yii\base\InvalidParamException
The file or directory to be published does not exist: /home/abhishek/Work/Yii/lpdapi/api/vendor/bower/jquery/dist
```
thing is it's looking at wrong place "/home/abhishek/Work/Yii/lpdapi/api/vendor/bower/jquery/dist"
it should not look for vendor  into `api` 
it should look at "/home/abhishek/Work/Yii/lpdapi/vendor/bower/jquery/dist"

Am I missing something in config?

I have configure my yii application separately
following is my config.

```
// Project directory 
.
├── api
│   └── controllers
├── composer.json
├── composer.lock
├── config
│   └── config.php
├── init
│   ├── assets
│   └── index.php
└── vendor
```

``` 
// composer.json
{
    "name": "lpdapi/mainproject",
    "require": {
        "yiisoft/yii2": ">=2.0.5"
    },
    "require-dev": {
        "yiisoft/yii2-gii": "*"
    },
    "authors": [
        {
            "name": "Abhishek Patil",
            "email": "abhishek@zeroth.me"
        }
    ]
}
```
```
//config/config.php
<?php

$config = [
    'id' => 'basic',
    'basePath' => dirname(__DIR__).'/api',
    'bootstrap' => ['gii'],
    'modules' => [
        'gii' => [
            'class' => 'yii\gii\Module',
        ],
        // ...
    ],
    'extensions' => require(__DIR__ . '/../vendor/yiisoft/extensions.php'),
];

return $config;

```
```
//init\index.php
<?php

// comment out the following two lines when deployed to production
defined('YII_DEBUG') or define('YII_DEBUG', true);
defined('YII_ENV') or define('YII_ENV', 'dev');

require(__DIR__ . '/../vendor/autoload.php');
require(__DIR__ . '/../vendor/yiisoft/yii2/Yii.php');

$config = require(__DIR__ . '/../config/config.php');

(new yii\web\Application($config))->run();
```

Fxp  is there 
```
composer global require "fxp/composer-asset-plugin:~1.1.1"
```
Vender dir
```
.
├── autoload.php
├── bin
│   ├── markdown -> ../cebe/markdown/bin/markdown
│   └── yii -> ../yiisoft/yii2/yii
├── bower-asset
│   ├── bootstrap
│   ├── jquery
│   ├── jquery.inputmask
│   ├── punycode
│   ├── typeahead.js
│   └── yii2-pjax
├── cebe
│   └── markdown
├── composer
│   ├── autoload_classmap.php
│   ├── autoload_files.php
│   ├── autoload_namespaces.php
│   ├── autoload_psr4.php
│   ├── autoload_real.php
│   ├── ClassLoader.php
│   ├── installed.json
│   └── LICENSE
├── ezyang
│   └── htmlpurifier
├── phpspec
│   └── php-diff
└── yiisoft
    ├── extensions.php
    ├── yii2
    ├── yii2-bootstrap
    ├── yii2-composer
    └── yii2-gii

```
here's how composer install  output looks like
```
composer install
Loading composer repositories with package information
Installing dependencies (including require-dev)
  - Installing yiisoft/yii2-composer (2.0.3)               
    Downloading: 100%         

  - Installing ezyang/htmlpurifier (v4.6.0)
    Downloading: 100%         

  - Installing bower-asset/jquery (2.1.4)
    Downloading: 100%         

  - Installing bower-asset/typeahead.js (v0.10.5)
    Downloading: 100%         

  - Installing phpspec/php-diff (v1.0.2)
    Downloading: 100%         

  - Installing bower-asset/bootstrap (v3.3.5)
    Downloading: 100%         

  - Installing bower-asset/yii2-pjax (v2.0.5)
    Downloading: 100%         

  - Installing bower-asset/punycode (v1.3.2)
    Downloading: 100%         

  - Installing bower-asset/jquery.inputmask (3.1.63)
    Downloading: 100%         

  - Installing cebe/markdown (1.1.0)
    Downloading: 100%         

  - Installing yiisoft/yii2 (2.0.6)
    Downloading: 100%         

  - Installing yiisoft/yii2-bootstrap (2.0.5)
    Downloading: 100%         

  - Installing yiisoft/yii2-gii (2.0.4)
    Downloading: 100%         

```

**Comment by SilverFire** [comment]
Yes, you missed the following in your composer.json

```json
    "extra": {
        "asset-installer-paths": {
            "npm-asset-library": "vendor/npm",
            "bower-asset-library": "vendor/bower"
        }
    }
```

**Comment by zeroth** [comment]
So I added the "extra" in my composer.json
then I deleted composer.lock and vendor dir 
then composer clear-cache
then composer install
but still the same error :( 
any other pointer ?

**Comment by SilverFire** [comment]
I have no ideas for now

**Comment by zeroth** [comment]
it looks like gii  by default assuming that vendor will be in basePath

**Comment by zeroth** [comment]
@SilverFire could you please try the same dir structure and config at your end and confirm this is not my systems problem ?

**Comment by zeroth** [comment]
Was just debugging and found that when Yii load JqueryAssets it gets the wrong path before that every other assets are getting correct path. here is the code for JqueryAssets.php
```
<?php
/**
 * @link http://www.yiiframework.com/
 * @copyright Copyright (c) 2008 Yii Software LLC
 * @license http://www.yiiframework.com/license/
 */

namespace yii\web;

/**
 * This asset bundle provides the [jquery javascript library](http://jquery.com/)
 *
 * @author Qiang Xue <qiang.xue@gmail.com>
 * @since 2.0
 */
class JqueryAsset extends AssetBundle
{
    public $sourcePath = '@bower/jquery/dist';
    public $js = [
        'jquery.js',
    ];
}
```
I have no clue how this thing works but I am guessing there is a problem with $sourcePath any suggestions ?

**Comment by zeroth** [comment]
So I found the solution,
I was kind of correct, 
after digging into Yii2's code I found,
```
public function getVendorPath()
    {
        if ($this->_vendorPath === null) {
            $this->setVendorPath($this->getBasePath() . DIRECTORY_SEPARATOR . 'vendor');
        }

        return $this->_vendorPath;
    }
```
it means if `vendorPath` is not given then `basePath` is consider as `vendorPath`

so just added `'vendorPath' => dirname(__DIR__).'/vendor',` in my config and all set.

I wish documentation for config could have been bit more explanatory about this.

@SilverFire  Thanks for the help.
